lib/instance.js

```diff
@@ -572,6 +572,36 @@ module.exports = (function() {
       }).then(function() {
         if (!options.fields.length) return this;
 
+        // create relational data for BelongsTo
+        var relationPromises = [];
+        Object.keys(this.dataValues).forEach(function(dataValue) {
+          if (_.has(self.options.includeMap, dataValue)) {
+            var includeModel = self.options.includeMap[dataValue].model;
```

Comment:
Hoist per iteration of loop.
I'm creating a new PR in Sequelize with this commit that will focus on just creating a shipping nested create method, lets continue the work there, pushing it in 45mins or so.